File: dataset/ourMethod/OPENAI_gpt-5-2025-08-07/Task185.cpp
#include <iostream>
#include <vector>
#include <stdexcept>
#include <limits>
#include <iomanip>
#include <algorithm>

double findMedianSortedArrays(const std::vector<int>& nums1, const std::vector<int>& nums2) {
    const std::vector<int>* A = &nums1;
    const std::vector<int>* B = &nums2;

    if (A->size() + B->size() == 0) {
        throw std::invalid_argument("Both input arrays are empty.");
    }

    if (A->size() > B->size()) {
        std::swap(A, B);
    }

    int m = static_cast<int>(A->size());
    int n = static_cast<int>(B->size());
    int low = 0, high = m;
    int half = (m + n + 1) / 2;

    while (low <= high) {
        int i = (low + high) / 2;
        int j = half - i;

        int Aleft  = (i == 0) ? std::numeric_limits<int>::min() : (*A)[i - 1];
        int Aright = (i == m) ? std::numeric_limits<int>::max() : (*A)[i];
        int Bleft  = (j == 0) ? std::numeric_limits<int>::min() : (*B)[j - 1];
        int Bright = (j == n) ? std::numeric_limits<int>::max() : (*B)[j];

        if (Aleft <= Bright && Bleft <= Aright) {
            if (((m + n) & 1) == 0) {
                long long leftMax = std::max(Aleft, Bleft);
                long long rightMin = std::min(Aright, Bright);
                return (leftMax + rightMin) / 2.0;
            } else {
                return static_cast<double>(std::max(Aleft, Bleft));
            }
        } else if (Aleft > Bright) {
            high = i - 1;
        } else {
            low = i + 1;
        }
    }

    throw std::invalid_argument("Input arrays must be sorted.");
}

static void runTest(const std::vector<int>& a, const std::vector<int>& b) {
    try {
        double res = findMedianSortedArrays(a, b);
        std::cout << std::fixed << std::setprecision(5) << res << "\n";
    } catch (const std::exception& ex) {
        std::cout << "Error: " << ex.what() << "\n";
    }
}

int main() {
    // 5 Test cases
    runTest({1, 3}, {2});                                       // 2.00000
    runTest({1, 2}, {3, 4});                                    // 2.50000
    runTest({}, {1});                                           // 1.00000
    runTest({0, 0}, {0, 0});                                    // 0.00000
    runTest({-1000000, -5, 0, 7}, {-3, 2, 2, 1000000});         // 1.00000
    return 0;
}

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task185.java
import java.util.*;

public class Task185 {
    public static double findMedianSortedArrays(int[] nums1, int[] nums2) {
        if (nums1.length > nums2.length) return findMedianSortedArrays(nums2, nums1);
        int m = nums1.length, n = nums2.length;
        int imin = 0, imax = m, half = (m + n + 1) / 2;
        while (imin <= imax) {
            int i = (imin + imax) / 2;
            int j = half - i;
            if (i < m && j > 0 && nums2[j - 1] > nums1[i]) {
                imin = i + 1;
            } else if (i > 0 && j < n && nums1[i - 1] > nums2[j]) {
                imax = i - 1;
            } else {
                int maxLeft;
                if (i == 0) maxLeft = nums2[j - 1];
                else if (j == 0) maxLeft = nums1[i - 1];
                else maxLeft = Math.max(nums1[i - 1], nums2[j - 1]);
                if (((m + n) % 2) == 1) return maxLeft;
                int minRight;
                if (i == m) minRight = nums2[j];
                else if (j == n) minRight = nums1[i];
                else minRight = Math.min(nums1[i], nums2[j]);
                return (maxLeft + minRight) / 2.0;
            }
        }
        return 0.0;
    }

    public static void main(String[] args) {
        int[] a1 = {1, 3}; int[] b1 = {2};
        int[] a2 = {1, 2}; int[] b2 = {3, 4};
        int[] a3 = {};     int[] b3 = {1};
        int[] a4 = {0, 0}; int[] b4 = {0, 0};
        int[] a5 = {2};    int[] b5 = {};

        double r1 = findMedianSortedArrays(a1, b1);
        double r2 = findMedianSortedArrays(a2, b2);
        double r3 = findMedianSortedArrays(a3, b3);
        double r4 = findMedianSortedArrays(a4, b4);
        double r5 = findMedianSortedArrays(a5, b5);

        System.out.printf("%.5f%n", r1);
        System.out.printf("%.5f%n", r2);
        System.out.printf("%.5f%n", r3);
        System.out.printf("%.5f%n", r4);
        System.out.printf("%.5f%n", r5);
    }
}